transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/BiosampleTransformations.scala

```diff
@@ -3,15 +3,42 @@ package org.broadinstitute.monster.encode.transformation
 import java.time.{LocalDate, OffsetDateTime}
 
 import org.broadinstitute.monster.encode.jadeschema.table.Biosample
+import org.slf4j.LoggerFactory
 import upack.Msg
 
 /** Transformation logic for ENCODE biosample objects. */
 object BiosampleTransformations {
   import org.broadinstitute.monster.common.msg.MsgOps
 
+  private val logger = LoggerFactory.getLogger(getClass)
+
   /** Transform a raw ENCODE biosample into our preferred schema. */
-  def transformBiosample(biosampleInput: Msg): Biosample = {
+  def transformBiosample(biosampleInput: Msg, joinedLibraries: Iterable[Msg]): Biosample = {
+    val biosample_id = CommonTransformations.readId(biosampleInput)
     val (auditLevel, auditLabels) = CommonTransformations.summarizeAudits(biosampleInput)
+
+    var product_ids = joinedLibraries
+      .flatMap(library => library.tryRead[String]("product_id"))
+      .toSet[String]
+    var lot_ids = joinedLibraries
+      .flatMap(library => library.tryRead[String]("lot_id"))
+      .toSet[String]
+
+    // check that there are not more than one unique product_id or lot_id.
+    // If there are multiple, default to None.
+    if (product_ids.size > 1) {
```

Comment:
You can push these `if`s down to the `Biosample.appy` call, and add `else` branches to make them expressions